.github/workflows/codeql.yml

```diff
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

Comment:
Maybe also ditch this?

Reply:
shouldn't we keep this? it would run when we make new PRs or should we keep it like
```yaml
on:
  pull_request:
```

Reply:
Maybe we used it wrong, but `pull_request` didn't trigger on pushes

Reply:
oh really? let me check that

Reply:
@askolesov BTW this is incorrect. `pull_request` **also** triggers on every push to the branch associated with the pull request